lib/request.js

```diff
@@ -128,7 +128,7 @@ module.exports = {
   set path(path) {
     const url = parse(this.req);
     url.pathname = path;
-    url.path = null;
+    url.path = url.pathname + url.search;
```

Comment:
maybe we should add `if (url.pathname === path) return` to avoid `stringify(url)`